pallets/aggregate/src/lib.rs

```diff
@@ -557,6 +616,85 @@ pub mod pallet {
                 state: self.state,
             });
         }
+
+        /// Return true iff the submitting account is allowed to add proofs to this domain, according
+        /// to the configured ProofSecurityRules.
+        fn can_add_proof(&self, account: &T::AccountId) -> bool {
+            match self.proof_rules {
+                ProofSecurityRules::Untrusted => true,
+                ProofSecurityRules::OnlyOwner => {
+                    account
+                        == self
+                            .owner
+                            .account()
+                            .expect("The domain does not have an owner; qed")
+                }
+                ProofSecurityRules::OnlyAllowlisted => {
+                    SubmittersAllowlist::<T>::get(self.id, account).is_some()
+                }
+            }
+        }
+
+        fn add_submitters(&mut self, submitters: &[AccountOf<T>]) -> Result<(), DispatchError> {
+            let count = submitters
+                .iter()
+                .filter(|s| !SubmittersAllowlist::<T>::contains_key(self.id, s))
+                .cloned()
+                .map(|s| SubmittersAllowlist::<T>::insert(self.id, s, ()))
+                .count();
+            self.increase_footprint_count(count.saturated_into())
+        }
+
+        fn remove_submitters(&mut self, submitters: &[AccountOf<T>]) -> Result<(), DispatchError> {
+            let count = submitters
+                .iter()
+                .filter(|s| SubmittersAllowlist::<T>::contains_key(self.id, s))
+                .cloned()
+                .map(|s| SubmittersAllowlist::<T>::remove(self.id, s))
+                .count();
+            self.decrease_footprint_count(count.saturated_into())
+        }
+
+        fn is_allowlist_empty(&self) -> bool {
+            use frame_support::StorageDoubleMap;
+            self.proof_rules != ProofSecurityRules::OnlyAllowlisted
+                || !SubmittersAllowlist::<T>::contains_prefix(self.id)
+        }
+
+        fn increase_footprint_count(&mut self, amount: u64) -> Result<(), DispatchError> {
```

Comment:
These two functions increase and decrease footprint count might use some dev documentation for future reference